src/common/actions/snaps.js

```diff
@@ -14,85 +11,37 @@ export const REMOVE_SNAP_ERROR = 'REMOVE_SNAP_ERROR';
 
 export function fetchUserSnaps(owner) {
   return async (dispatch) => {
-    const url = `${BASE_URL}/api/launchpad/snaps/list`;
     const query = `owner=${encodeURIComponent(owner)}`;
 
     dispatch({
-      type: FETCH_SNAPS
-    });
-
-    try {
-      const response = await fetch(`${url}?${query}`, {
-        headers: { 'Content-Type': 'application/json' },
-        credentials: 'same-origin'
-      });
-      await checkStatus(response);
-      const result = await response.json();
-      if (result.status !== 'success') {
-        throw getError(response, result);
+      type: FETCH_SNAPS,
+      [CALL_API]: {
+        types: [ FETCH_SNAPS_SUCCESS, FETCH_SNAPS_ERROR ],
```

Comment:
If I read this correctly, we are not doing the fetch in the action creator anymore, but we dispatch an action with API call meta data that is ignored by reducers, but middleware uses it to make the call and dispatch another actions based on the results?

That's quite a confusing abstraction layer when I first look at it. And I wonder what benefits does it have over some helper/wrapper around fetch that would add common headers and token to the request.

As far as I can see also there is no way to do anything more with API result other then dispatching an action with it?